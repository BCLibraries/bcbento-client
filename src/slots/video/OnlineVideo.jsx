import React from 'react';
import blankScreen from "./blank-screen.svg";

function OnlineVideo({video}) {
    const screen = (video.covers && video.covers.length > 0) ? video.covers[0] : blankScreen;
    const screenClass = (video.covers && video.covers.length > 0) ? 'online-video-thumb__screenshot' : 'online-video-thumb__blank_screen';

    return <li className="online-video col-md-4">
        <a href={video.getit}>
            <div className="online_video__thumb">
                <img src={screen} alt={`Watch ${video.title} online`} className={screenClass}/>
            </div>
        </a>

        <h3 className="online-video__media-heading media-heading">
            <a href={video.link} className="online-video__title">{video.title}</a>
        </h3>

        {video.date}

        <div className="online-video__creator">{creatorName(video)}</div>

        <div className="online-video__watch-link"><a href={video.getit}>Watch online</a></div>
    </li>
}

function creatorName(video) {
    if (video.creator) {
        return video.creator;
    }

    if (video.contributors[0]) {
        return video.contributors[0];
    }

    return '';
}

export default OnlineVideo;