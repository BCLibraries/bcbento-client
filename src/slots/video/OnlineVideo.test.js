import renderer from 'react-test-renderer';
import OnlineVideo from './OnlineVideo.jsx';

it('renders correctly', () => {
    const video = {
        "id": "ALMA-BC51493661490001021",
        "title": "Do unto otters",
        "date": "[2012]",
        "type": "Video",
        "creator": null,
        "contributors": [
            "Paul R. Gagne 1956- film producer",
            "Melissa Reilly Ellard film producer",
            "Galen Fott adapter, film director",
            "Jack Sundrud actor",
            "Rusty Young actor",
            "Diana Canova actor",
            "Laurie Keller author",
            "Weston Woods Studios, production company",
            "Scholastic Inc., production company"
        ],
        "coverImages": [
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9780531208922&issn=/MC.JPG&client=primo",
            },
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9780531208922&issn=/MC.JPG&client=primo",
            },
            {
                "url": "https://proxy-na.hosted.exlibrisgroup.com/exl_rewrite/syndetics.com/index.aspx?isbn=9780531208922&issn=/MC.JPG&client=primo",
            }
        ],
        "available": null,
        "isPhysical": false,
        "isElectronic": true,
        "linkableId": "ALMA-BC51493661490001021",
        "screenCap": null,
        "format": "1 streaming video file (12 min.) : digital, WMV file, sound, color.",
        "mms": "99137930178901021",
        "availability": null,
    };

    const component = renderer.create(
        <OnlineVideo video={video}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
