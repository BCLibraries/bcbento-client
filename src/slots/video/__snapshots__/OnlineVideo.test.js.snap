// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="online-video"
>
  <a
    href="https://mlib.bc.edu/reserves-api/items/99137930178901021"
  >
    <div
      className="online_video__thumb"
    >
      <img
        alt="Watch Do unto otters online"
        className="online-video-thumb__blank_screen"
        onError={[Function]}
        src="https://library.bc.edu/images/blank-screen.svg"
      />
    </div>
  </a>
  <div>
    <div
      className="catalog-result-item__type"
    >
      Video
    </div>
    <h3
      className="online-video__media-heading media-heading"
    >
      <a
        className="online-video__title"
        href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC51493661490001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
      >
        Do unto otters
      </a>
    </h3>
  </div>
  [2012]
  <div
    className="online-video__creator"
  >
    Paul R. Gagne 1956- film producer
  </div>
  <div
    className="online-video__watch-link"
  >
    <a
      href="https://mlib.bc.edu/reserves-api/items/99137930178901021"
    >
      Watch online
    </a>
  </div>
</div>
`;
